python_modules/dagster/dagster/_core/definitions/auto_materialize_reason.py

```diff
@@ -0,0 +1,75 @@
+from enum import Enum
+from typing import NamedTuple
+
+
+class AutoMaterializeResult(Enum):
+    """Represents the set of results of the auto-materialize logic.
+
+    MATERIALIZE: The asset should be materialized by a run kicked off on this tick
+    SKIP: The asset should not be materialized by a run kicked off on this tick, because future
+        ticks are expected to materialize it.
+    DISCARD: The asset should not be materialized by a run kicked off on this tick, but future
+        ticks are not expected to materialize it.
+    """
+
+    MATERIALIZE = "MATERIALIZE"
+    SKIP = "SKIP"
+    DISCARD = "DISCARD"
+
+
+class AutoMaterializeCondition(Enum):
+    """Represents the set of conditions that can impact auto-materialization for an asset."""
+
+    def __init__(self, id: str, result: AutoMaterializeResult):
+        self.id = id
+        self.result = result
+
+    # Materialization Reasons
+    FRESHNESS = ("FRESHNESS", AutoMaterializeResult.MATERIALIZE)
+    DOWNSTREAM_FRESHNESS = ("DOWNSTREAM_FRESHNESS", AutoMaterializeResult.MATERIALIZE)
+    PARENT_MATERIALIZED = ("PARENT_MATERIALIZED", AutoMaterializeResult.MATERIALIZE)
+    MISSING = ("MISSING", AutoMaterializeResult.MATERIALIZE)
+
+    # Skip Reasons
+    PARENT_OUTDATED = ("PARENT_OUTDATED", AutoMaterializeResult.SKIP)
+
+    # Discard Reasons
+
+
+class AutoMaterializeConditionReason(NamedTuple):
+    """Denotes the reason that the auto-materialize logic decided that an asset should be materialized.
```

Comment:
I'm finding this a little difficult to parse. Is this class only relevant in the case that the `AutoMaterializeResult` of the `AutoMaterializeCondition` is MATERIALIZE? From the name, I would expect it to apply to any `AutoMaterializeCondition`.

Reply:
I just forgot to update this docstring -- this is just a wrapper class that allows (in the future) more specific information to be attached to the underlying condition

Reply:
e.g. the "condition" is "DOWNSTREAM_FRESHNESS", and the specific data is "Assets X, Y, and Z needed fresher data", or the "condition" is "PARENT_NEVER_MATERIALIZED", and the specific data is "Upstream asset A has never been materialized"